main.py

```diff
@@ -1,102 +1,40 @@
-    elif Actions.GAME_WON in events:
-        paused = True
-        screens["complete"].enabled = True
-    elif Actions.GAME_OVER in events:
-        paused = True
-        screens["gameover"].enabled = True
-    #toggle relevent ui screens
-    elif Actions.TOGGLE_PAUSE in events:
-        paused = not paused
-        screens["pause"].enabled = not screens["pause"].enabled
-    elif Actions.TOGGLE_INVENTORY in events:
-        paused = not paused
-        screens["pause"].enabled = not screens["pause"].enabled
-        screens["inventory"].enabled = not screens["inventory"].enabled
-    elif Actions.TOGGLE_SHOW_FPS in events:
-        show_framerate = not show_framerate
-
-    delta = clock.tick(MAX_FPS) / 1000.0
-    if show_framerate:
-        ui_overlay.text_elements["framerate"].value = "{0} fps".format(int(clock.get_fps()))
-    else:
-        ui_overlay.text_elements["framerate"].value = ""
-    ui_overlay.update(delta)
-
-    #render the game field, a delta of 0 means don't do any physics updates, events of [] means dont perform any inputs
-    if paused:
-        game.doFrame(surface, 0, [])
-    else:
-        game.doFrame(surface, delta, events)
-
-    #display any active ui screens
-    for screen in screens.values():
-        screen.doFrame(surface, delta, events)
-
-    #render the app-scope UI
-    for (label, position, _) in ui_overlay.get_drawables():
-        surface.blit(label, position)
-
-    #give it to the user
-    pygame.display.update()
+from game_loop import Game
+from argparse import ArgumentParser
+from curses import wrapper
+
+def main(_):
+    parser = ArgumentParser(description="TODO: Add a description")
+    parser.add_argument('--settings', default='settings.cfg', help="Location of the settings file to load")
+    parser.add_argument('--show_fps', action='store_true', help="If enabled, the fps ticker will be shown by default")
+    parser.add_argument('--max_fps', type=int, help="The maximum framerate of the game")
+    parser.add_argument('--width', type=int, help="Width of the screen")
+    parser.add_argument('--height', type=int, help="Height of the screen")
+    parser.add_argument('--debug', action='store_true', help="If enabled, stdin will function as a full debug console")
+
+    args = parser.parse_args()
+
+    config_handle = open(args.settings, "r")
+    config = json.load(config_handle)
+    config_handle.close()
+
+    if args.max_fps is not None:
+        config['max_fps'] = args.max_fps
+    if args.width is not None:
+        config['width'] = args.width
+    if args.height is not None:
+        config['height'] = args.height
+    if args.show_fps:
+        config['show_fps'] = args.show_fps
+    if args.debug:
+        config['use_debug_console'] = args.debug
+
+    game = Game(config)
+    if config.get("use_debug_console", False):
+        from debug_console import DebugConsole
+        game = DebugConsole(game)
+
+    while game.is_running:
+        game.run_loop()
+
+wrapper(main)
```

Comment:
The wrapper function is from the curses library. It will switch to a new terminal context so that the readline library in debug_console doesn't screw with settings.